config/v1/types_infrastructure.go

```diff
@@ -600,9 +600,202 @@ type OvirtPlatformStatus struct {
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Datacenters []string `json:"datacenters"`
+}
+
+// VSpherePlatformDeploymentSuitable
+// +kubebuilder:validation:Enum=Allowed;Disallowed
+type VSpherePlatformDeploymentSuitable string
+
+const (
+	// Allowed indicates that the Deployment Zone is suitable for
+	// control plane nodes.
+	Allowed VSpherePlatformDeploymentSuitable = "Allowed"
+
+	// Disallowed indicates that the Deployment Zone is not suitable for
+	// control plane nodes.
+	Disallowed VSpherePlatformDeploymentSuitable = "Disallowed"
+)
+
+// VSpherePlatformNodeNetworkingSpec holds the network CIDR(s) and port group name for
+// including and excluding IP ranges in the cloud provider.
+// This would be used for example when multiple network adapters are attached to
+// a guest to help determine which IP address the cloud config manager should use
+// for the external and internal node networking.
+type VSpherePlatformNodeNetworkingSpec struct {
+	// networkSubnetCidr IP address on VirtualMachine's network interfaces included in the fields' CIDRs
+	// that will be used in respective status.addresses fields.
+	// +optional
+	NetworkSubnetCIDR []string `json:"networkSubnetCidr,omitempty"`
+
+	// network VirtualMachine's VM Network names that will be used to when searching
+	// for status.addresses fields. Note that if internal.networkSubnetCIDR and
+	// external.networkSubnetCIDR are not set, then the vNIC associated to this network must
+	// only have a single IP address assigned to it.
+	// The available networks (port groups) can be listed using
+	// govc ls 'network/*'
+	// +optional
+	Network string `json:"network,omitempty"`
+
+	// excludeNetworkSubnetCidr IP addresses in subnet ranges will be excluded when selecting
+	// the IP address from the VirtualMachine's VM for use in the status.addresses fields.
+	// +optional
+	ExcludeNetworkSubnetCIDR []string `json:"excludeNetworkSubnetCidr,omitempty"`
+}
+
+// VSpherePlatformNodeNetworking holds the external and internal node networking spec.
+type VSpherePlatformNodeNetworking struct {
+	// external represents the VSpherePlatformNodeNetworkingSpec of the node that is externally routable.
+	// +optional
+	External VSpherePlatformNodeNetworkingSpec `json:"external"`
+	// internal represents the VSpherePlatformNodeNetworkingSpec of the node that is routable only within the cluster.
+	// +optional
+	Internal VSpherePlatformNodeNetworkingSpec `json:"internal"`
+}
+
 // VSpherePlatformSpec holds the desired state of the vSphere infrastructure provider.
-// This only includes fields that can be modified in the cluster.
-type VSpherePlatformSpec struct{}
+// In the future the cloud provider operator, storage operator and machine operator will
+// use these fields for configuration.
+type VSpherePlatformSpec struct {
+	// vcenters holds the connection details for services to communicate with vCenter.
+	// Currently, only a single vCenter is supported.
```

Comment:
What happens if no vCenter is provided?

Reply:
@JoelSpeed not quite sure how to handle this case.
Existing clusters won't have vcenters defined but new clusters will.
It should be required with the min and max as of today set to 1.

Reply:
What will the consumers of this API (storage and ccm?) do if there isn't anything defined here? Do we fall back to the vCenter defined in the configmap in openshift-config? I would document whatever the fallback is